src/Pode.psm1

```diff
@@ -1,38 +1,98 @@
+<#
+.SYNOPSIS
+    Pode PowerShell Module
+
+.DESCRIPTION
+    This module sets up the Pode environment, including
+    localization and loading necessary assemblies and functions.
+
+.PARAMETER UICulture
+    Specifies the culture to be used for localization.
+
+.EXAMPLE
+    Import-Module -Name "Pode" -ArgumentList @{ UICulture = 'ko-KR' }
+    Sets the culture to Korean.
+
+.EXAMPLE
+    Import-Module -Name "Pode"
+    Uses the default culture.
+
+.EXAMPLE
+    Import-Module -Name "Pode" -ArgumentList 'it-SM'
+    Uses the Italian San Marino region culture.
+
+.NOTES
+    This is the entry point for the Pode module.
+#>
+
+param(
+    [string]$UICulture
+)
+
 # root path
 $root = Split-Path -Parent -Path $MyInvocation.MyCommand.Path
+# Import localized messages
+if ([string]::IsNullOrEmpty($UICulture)) {
+    $UICulture = $PsUICulture
+}
+
+#Culture list available here https://azuliadesigns.com/c-sharp-tutorials/list-net-culture-country-codes/
+Import-LocalizedData -BindingVariable tmpMsgtable -BaseDirectory (Join-Path -Path $root -ChildPath 'Locales') -UICulture $UICulture -ErrorAction:SilentlyContinue
+if ($null -eq $tmpMsgtable) {
+    try {
+        Import-LocalizedData -BindingVariable tmpMsgtable -BaseDirectory (Join-Path -Path $root -ChildPath 'Locales') -UICulture 'en' -ErrorAction:Stop
+    }
+    catch {
+        throw
+    }
+}
+
+try {
+    # Create the global msgTable read-only variable
+    New-Variable -Name 'msgTable' -Value $tmpMsgtable -Scope Global -Option ReadOnly -Force
 
-# load assemblies
-Add-Type -AssemblyName System.Web
-Add-Type -AssemblyName System.Net.Http
+    # load assemblies
+    Add-Type -AssemblyName System.Web -ErrorAction Stop
+    Add-Type -AssemblyName System.Net.Http -ErrorAction Stop
 
-# Construct the path to the module manifest (.psd1 file)
-$moduleManifestPath = Join-Path -Path $root -ChildPath 'Pode.psd1'
+    # Construct the path to the module manifest (.psd1 file)
+    $moduleManifestPath = Join-Path -Path $root -ChildPath 'Pode.psd1'
 
-# Import the module manifest to access its properties
-$moduleManifest = Import-PowerShellDataFile -Path $moduleManifestPath
+    # Import the module manifest to access its properties
+    $moduleManifest = Import-PowerShellDataFile -Path $moduleManifestPath -ErrorAction Stop
+}
+catch {
+    throw
+}
```

Comment:
Would the `-ErrorAction Stop` parameters not already throw and exit the script? (same for the catch/throw below as well

Reply:
I thought that was the case. But I was wrong.

To make the script fail when Pode.psm1 fails, I have to add a throw and call the import-module like this example:
```
try{
	Import-Module -Name 'Pode'
}catch {
	throw
}
```